### PR TITLE
add top-level retry-logic

### PR DESCRIPTION
**Pull request by mwhooker**
While aws-sdk has its own retry logic, there are certain cases that it doesn't
cover. We'll add a retrier at the top-level to catch those cases.

Should resolve the "unexpected EOF" errors we've been seeing.

https://github.com/aws/aws-sdk-go/issues/466
https://github.com/aws/aws-sdk-go/issues/675

### CODE REVIEW

**Comment by mwhooker** [comment]
rebased on top of #31

File: mhook.go [hunk 1]
```diff
@@ -358,9 +378,12 @@ var (
...
 			fmt.Printf("Downloading from %s\n", *mhook.Key(target))
-			if err := mhook.Download(target, destination); err != nil {
-				// temporary debug line
-				fmt.Printf("mhook err %#v\n", err)
+			if c.Int("retries") < 1 {
+				return fmt.Errorf("Retries must be greater than 0")
+			}
+			re := &retryer{c.Int("retries")}
+
+			if err := re.Retry(func() error { return mhook.Download(target, destination) }); err != nil {
```
Comment (hatchan) [inline mhook.go]: Maybe a bit nitpicky, but maybe make it `Try`

**Comment by hatchan** [comment]
:+1:

**Comment by termie** [comment]
@mwhooker when you've decided whether or not to change the name pls merge and get this into use, are we using a ferengi branch on ferengi? if so that needs the patch too

**Comment by mwhooker** [comment]
see also
* https://github.com/grammarly/rocker/blob/d0220cddfa1bcd107db138498c0f4de5d9f0ecc2/src/rocker/storage/s3/retryer.go#L65
* https://github.com/grammarly/rocker/blob/d0220cddfa1bcd107db138498c0f4de5d9f0ecc2/src/rocker/storage/s3/s3.go#L172

**Pull request merged**